api/server/runner_httptrigger_test.go

```diff
@@ -611,3 +615,61 @@ func TestTriggerRunnerMinimalConcurrentHotSync(t *testing.T) {
 		}
 	}
 }
+
+//Generally we want to return the user function status, only making an adjustment to
+//502 for 5xx codes. Otherwise, we return the code that our service has selected.
+//Failing any information, we just give it a 200.
+func TestFinalStatusSetting(t *testing.T) {
```

Comment:
Kind of feel these tests would be easier to read if they were case based. 

Given user error of X and function error of Y : expect endpoint error of Z

Reply:
I disagree on this one. Writing out the properties we want from the code is way clearer than the case based style where you have to translate it all and all the check logic is smudged together.

This style has already proved it's worth to my mind (with the config equality bug), and even here they helped steer the code nicely.